package_nso_to_oc/xe/xe_system.py

```diff
@@ -146,6 +152,47 @@ def xe_system_services(config_before: dict, config_leftover: dict) -> None:
         del config_leftover["tailf-ned-cisco-ios:ip"]["http"]["server"]
     else:
         openconfig_system_services["openconfig-system-ext:http"]["openconfig-system-ext:config"]["openconfig-system-ext:http-enabled"] = True
+        del config_leftover["tailf-ned-cisco-ios:ip"]["http"]["server"]
+    # IP http secure server
+    if config_before.get("tailf-ned-cisco-ios:ip", {}).get("http", {}).get("secure-server", True) is False:
+        openconfig_system_services["openconfig-system-ext:http"]["openconfig-system-ext:config"][
+            "openconfig-system-ext:https-enabled"] = False
+        del config_leftover["tailf-ned-cisco-ios:ip"]["http"]["secure-server"]
+    else:
+        openconfig_system_services["openconfig-system-ext:http"]["openconfig-system-ext:config"][
+            "openconfig-system-ext:https-enabled"] = True
+        del config_leftover["tailf-ned-cisco-ios:ip"]["http"]["secure-server"]
+    # IP http max-connections
```

Comment:
Since you only care about booleans and either case is guaranteed to run, you can simplify to something like this.

```suggestion
      openconfig_system_services["openconfig-system-ext:http"]["openconfig-system-ext:config"][
          "openconfig-system-ext:https-enabled"] = config_before.get("tailf-ned-cisco-ios:ip", {})
              .get("http", {}).get("secure-server", True)
      del config_leftover["tailf-ned-cisco-ios:ip"]["http"]["secure-server"]
```

Reply:
Although, having that else clause makes me wonder, is it guaranteed that `config_leftover["tailf-ned-cisco-ios:ip"]["http"]["secure-server"]` will always exist? Because if it doesn't, python will raise an error when attempting delete.